### ISSUE TITLE
Configure navigation bar action buttons to be shown in an overflow menu in Android

### ISSUE THREAD

**Issue by varungupta85** [exponentjs/ex-navigation#232]
I am currently using `ex-navigator` in my app and considering moving to `ex-navigation` because of its enhanced API and am also looking for a more native feel with Android. `ex-navigation` provides Drawer out of the box which is good. I wanted to know if there are any plans of adding some configuration to the route such that the navigation bar buttons could be configured to be shown in an overflow menu on the right which appears when the user presses the overflow icon in Android. Such a functionality is present in `ToolbarAndroid` but it is kind of tricky to integrate `ToolbarAndroid` with this. If someone already has a solution for it, I would really appreciate it if he/she can share it or at least provide some hints on how to achieve it.

**Comment by chirag04** [comment]
exnavigation makes no assumption about the navigation bar buttons and gives you `renderRight`. Inside `renderRight` of your screen route config, you can do all sort of things like overflow menu etc.

exnavigation does not use `ToolbarAndroid` but lets you pass in your own header component. You can create a custom Header Component which renders ToolbarAndroid under the hood.

I def think what you describe is a very common pattern on android and would be nice to add to exnavigation. PR would be awesome.

cc @skevy @brentvatne

**Comment by varungupta85** [comment]
@chirag04 Thanks for the lightning fast response 👍  . I realized that I could use `renderRight` as you have suggested to achieve something like this but was not sure on how to show the overflow menu once the user clicks the overflow icon. I suppose it would be some kind of modal with transparent background or is there a better way of doing that. I didn't know about the header component you mentioned. I will give it a try.

**Comment by chirag04** [comment]
by overflow menu you mean the context menu on android i suppose. https://material.google.com/components/menus.html#menus-usage

Here is my implementation of context menu/popup menu/overflow menu.

**Comment by varungupta85** [comment]
Hi @chirag04, I meant the context menu as shown in the link you shared. Your comment ended with 'Here is my implementation...'. I am not sure if you meant to share some code snippet.

**Comment by chirag04** [comment]
haha. I meant to share the link. here it is: https://github.com/facebook/react-native/issues/3004#issuecomment-202598006

**Comment by vonovak** [comment]
I'm using custom navbar on Android that is a toolbarAndroid from RN. The only thing I don't like so much is that ex-navigation doesn't provide a nice built-in  way to specify different actions that woukd display under the overflow menu.

**Comment by chirag04** [comment]
@vonovak can you share the implementation of navbar with toolbarAndroid? We can probably add it to the readme. Seems like a lot of people would want to do this on android.

**Comment by vonovak** [comment]
@chirag04 sure, I bet a lot of people will want to do this. And +1 on improving the docs any way possible. However, I think this needs a bit more thinking from someone who understands the internals of ex-navigation on how exactly the navbar should be configured because I'm not 100% happy with my current setup. 
Here is a gist: https://gist.github.com/vonovak/81d33b0fa55a5f77e9169d143e4293eb

**Comment by chirag04** [comment]
Nice. Thanks for sharing. I haven't played around android and exnavigation enough but i will do it soon. In terms of communication between screen and navbar, your setup looks fine. there is an event emitter also which can be useful at time. @skevy would have a lot more thoughts on this than me.

**Comment by vonovak** [comment]
Yes, I know about the event emitter, but personally I find it overly complicated. What I like better is passing an object as a prop, and having both navbar and the screen manipulate it directly. Since this doesn't seem to be possible, there is the option of saving the prop into a store - the prop is then passed to screen without problem, and the navbar can access it through the injected store. The only trouble now is you need that store :D well...
